src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -458,11 +459,11 @@ private void writeData(DataFrame data, Sheet sheet, int startRow, int startCol,
                             setMissing(cell);
                         else {
                             if(Double.isInfinite(doubleValues[j])) {
-                              cell.setCellType(CellType.ERROR);
+
                               cell.setCellErrorValue(FormulaError.NA.getCode());
                             } else {
                               cell.setCellValue(doubleValues[j]);
-                              cell.setCellType(CellType.NUMERIC);
```

Comment:
Same here. Is the cell type set implicitly by serCellValue?

Reply:
Correct, see https://poi.apache.org/apidocs/4.1/org/apache/poi/ss/usermodel/Cell.html#setCellType-org.apache.poi.ss.usermodel.CellType-